components/box/Box.js

```diff
@@ -0,0 +1,84 @@
+import React, { PureComponent } from 'react';
+import PropTypes from 'prop-types';
+import cx from 'classnames';
+import theme from './theme.css';
+
+const spacings = [0, 1, 2, 3, 4, 5, 6, 7, 8];
+
+class Box extends PureComponent {
+  static propTypes = {
+    children: PropTypes.node,
+    className: PropTypes.string,
+    element: PropTypes.node,
+    margin: PropTypes.oneOf(spacings),
+    marginBottom: PropTypes.oneOf(spacings),
+    marginLeft: PropTypes.oneOf(spacings),
+    marginRight: PropTypes.oneOf(spacings),
+    marginTop: PropTypes.oneOf(spacings),
+    padding: PropTypes.oneOf(spacings),
+    paddingBottom: PropTypes.oneOf(spacings),
+    paddingLeft: PropTypes.oneOf(spacings),
+    paddingRight: PropTypes.oneOf(spacings),
+    paddingTop: PropTypes.oneOf(spacings),
+  };
+
+  static defaultProps = {
+    element: 'div',
+    margin: 0,
+    marginBottom: 0,
+    marginLeft: 0,
+    marginRight: 0,
+    marginTop: 0,
+    padding: 0,
+    paddingBottom: 0,
+    paddingLeft: 0,
+    paddingRight: 0,
+    paddingTop: 0,
+  };
+
+  render() {
+    const {
+      children,
+      className,
+      element,
+      margin,
+      marginBottom,
+      marginLeft,
+      marginRight,
+      marginTop,
+      padding,
+      paddingBottom,
+      paddingLeft,
+      paddingRight,
+      paddingTop,
+      ...others
+    } = this.props;
+
```

Comment:
**Q:** Would it make sense to drop the `margin-#` classes and solely rely on the `margin-{direction}-#` classes?

This way, you could set the default values when destructuring here and remove the `margin-#` classes all together. This way we can add a little bit more sugar, eg. `marginVertical` and `marginHorizontal` (same shorthand like you would use in regular CSS)

Blatantly copied from `este`:

```js
const {
    margin,
    marginHorizontal = margin, // ES6 default values ftw?
    marginVertical = margin,
    marginBottom = marginVertical,
    marginLeft = marginHorizontal,
    marginRight = marginHorizontal,
    marginTop = marginVertical,

    padding,
    paddingHorizontal = padding,
    paddingVertical = padding,
    paddingBottom = paddingVertical,
    paddingLeft = paddingHorizontal,
    paddingRight = paddingHorizontal,
    paddingTop = paddingVertical,
} = this.props;
```

This way, you can also remove a lot of the defaultProps, as they will all fall back to `margin` 👍